Source: https://github.com/gmtsciencedev/bioit-dockers

A simple motus docker.

# Database

The database is not included in the image you must download it manually:

```bash
docker run -it --rm -v /some/path:/resource gmtscience/motus sh -c 'motus downloadDB && mv /opt/conda/lib/python3.9/site-packages/motus/db_mOTU /resource/'
cd /some/path
tar cvzf motus.tgz motus
```

The last line is intended for scitq usage so that the resource can be pushed to some S3 storage and loaded with `-r 's3://bucket/resource/motus.tgz|untar'`.


# Usage

A typical use for motus:

```bash
docker run -it -v /some/path:/resource -v /output:/output -v /input:/input \
    gmtscience/motus
    sh -c "motus profile -db /resource/db_mOTU /input/myseq.fq -e -o /output/myseq.motus -t $(nproc)"
```

## command line detail

motus uses python argparse in a non-standard way which prevents command line help to be displayed when database is not downloaded, so just in case, here it is:

```text
Program: motus - a tool for marker gene-based OTU (mOTU) profiling
Version: 3.0.3
Reference: Milanese et al. Microbial abundance, activity and population genomic profiling with mOTUs2. Nature Communications (2019). doi: 10.1038/s41467-019-08844-4

Usage: motus <command> [options]

Command:
      downloadDB  Download the mOTUs database

-- Taxonomic profiling
      profile     Perform taxonomic profiling (map_tax + calc_mgc + calc_motu) in a single step
      merge       Merge several taxonomic profiling results into one table

      map_tax     Map reads to the marker gene database
      calc_mgc    Calculate marker gene cluster (MGC) abundance
      calc_motu   Summarize MGC abundances into a mOTU profile

      prep_long   Prepare long reads to be sequenced by mOTUs

-- SNV calling
      map_snv     Map reads to the marker gene database for SNV calling
      snv_call    Generate SNV profiles (using metaSNV)

Type motus <command> to print the help menu for a specific command
```

```text
(base) root@09230d6abbcc:/tmp# motus -db /data/ref/motus3/db_mOTU profile 

Usage: motus profile [options]

Input options:
   -f  FILE[,FILE]  input file(s) for reads in forward orientation, fastq(.gz)-formatted
   -r  FILE[,FILE]  input file(s) for reads in reverse orientation, fastq(.gz)-formatted
   -s  FILE[,FILE]  input file(s) for unpaired reads, fastq(.gz)-formatted
   -n  STR          sample name ['unnamed sample']
   -i  FILE[,FILE]  provide SAM or BAM input file(s)  (generated by motus map_tax)
   -m  FILE         provide a mgc reads count file (generated by motus calc_mgc)
   -db DIR          provide a different database directory

Output options:
   -o  FILE         output file name [stdout]
   -I  FILE         save the result of BWA in BAM format (output of motus map_tax)
   -M  FILE         save the mgc reads count (output of motus calc_mgc)
   -e               only species with reference genomes (ref-mOTUs)
   -u               print the full name of the species
   -c               print result as counts instead of relative abundances
   -p               print NCBI taxonomy identifiers
   -B               print result in BIOM format
   -C  STR          print result in CAMI format (BioBoxes format 0.9.1)
                    Values: [precision, recall, parenthesis]
   -q               print the full rank taxonomy
   -A               print all taxonomic levels together (kingdom to mOTUs, override -k)
   -k  STR          taxonomic level [mOTU]
                    Values: [kingdom, phylum, class, order, family, genus, mOTU]

Algorithm options:
   -g  INT          number of marker genes cutoff: 1=higher recall, 6=higher precision [3]
   -l  INT          min length of the alignment (bp) [75]
   -t  INT          number of threads [1]
   -v  INT          verbosity level: 1=error, 2=warning, 3=message, 4+=debugging [3]
   -y  STR          type of read counts [insert.scaled_counts]
                    Values: [base.coverage, insert.raw_counts, insert.scaled_counts]

(base) root@09230d6abbcc:/tmp# motus -db /data/ref/motus3/db_mOTU merge

Usage: motus merge [options]

Input options:
   -i FILE[,FILE] list of mOTU profiles to merge (comma separated)
   -d DIR         merge all files in the directory DIR
   -a STR[,STR]   add pre-computed profiles from different environmental samples
                  Values: [all, air, bioreactor, bee, cat,
                  cattle, chicken, dog, fish, freshwater, human,
                  marine, mouse, pig, sheep, soil, termite, wastewater]

Output options:
   -o FILE        output file name [stdout]
   -B             print result in BIOM format

Algorithm options:
   -v INT         verbosity level: 1=error, 2=warning, 3=message, 4+=debugging [3]
```
